### ISSUE TITLE
Multi-threaded option

### ISSUE THREAD

**Issue by NedkoPNedev** [davidmoten/big-sorter#9]
Hello David!

First of all, I would like to thank you for this repo, I've been using it for some time for a university project of mine.

I would like to ask is it possible for big-sorter to become multi-threaded (probably in the step where you sort the split chunks before merging)? 

Also, is there a way to speed up the process by massaging some parameters before sorting? I saw these maxItemsPerFile, maxFilesPerMerge but not sure how to standardize the parameters to work efficiently for files of different sizes.

Thanks a lot!

Regards, Nedko

**Comment by davidmoten** [comment]
Glad it's been useful. I've just added a commit that allows you to specify that `Arrays.parallelSort` is used for the initial sort. I needed to copy `java.util.ArrayList` and `java.util.AbstractList` into the codebase so that I could add a `parallelSort` method that avoided one array copy but it works fine. You'll need to play with `maxItemsPerFile` (which is about how big the files are for the initial split) to see at what point an initial parallel sort becomes useful. By using this I reduced a sort in time by about 15%.

The values that you use for maxItemsPerFile (which defaults to 100,000) and for maxFilesPerMerge (which defaults to 100) will depend on the characteristics of your data, cpu and io system so you'll need to tune it yourself. SSD will behave very differently from Hard Disk for instance for the io system. Are you using SSD for the sort?

I'll release this shortly, I'll just check the other issue that someone raised.
